### ISSUE TITLE
incorrect T extends tilelayer ?

### ISSUE THREAD

**Issue by AdrienLemaire** [Chnapy/tiled-types#8]
Hi @Chnapy .

I'm trying to type a list of objects with properties and gids.
Looking at your types, I found:
```ts
export type TiledMapType = 'orthogonal' | 'isometric' | 'staggered' | 'hexagonal';

export interface TiledObject<T extends TiledMapType> {
    ellipse?: boolean;
    gid: T extends 'tilelayer' ? number : never;
    ...
}
```

tsserver always returns an error `Type 'number' is not assignable to type 'never'` 

How can T extends 'tilelayer'  if T extends TiledMapType  ?
Is that a bug in the types ?

**Comment by Chnapy** [comment]
Yes, I'll simplify the type like that `gid?: number;`

**Comment by Chnapy** [comment]
Fix published, please upgrade to `1.3.0` 👍
